### PR TITLE
[Doc] Refine algorithm readme with model performance table

### PR DESCRIPTION
**Pull request by ly015**
## Motivation

Update algorithm readme with model performance table and page links for algorithms for better user experience.

## Checklist

**Before PR**:

- [ ] I have read and followed the workflow indicated in the [CONTRIBUTING.md](https://github.com/open-mmlab/mmpose/blob/master/.github/CONTRIBUTING.md) to create this PR.
- [ ] Pre-commit or linting tools indicated in [CONTRIBUTING.md](https://github.com/open-mmlab/mmpose/blob/master/.github/CONTRIBUTING.md) are used to fix the potential lint issues.
- [ ] Bug fixes are covered by unit tests, the case that causes the bug should be added in the unit tests.
- [ ] New functionalities are covered by complete unit tests. If not, please add more unit tests to ensure correctness.
- [ ] The documentation has been modified accordingly, including docstring or example tutorials.

**After PR**:

- [ ] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

File: configs/body_2d_keypoint/topdown_heatmap/README.md [hunk 1]
```diff
@@ -5,3 +5,73 @@ Top-down methods divide the task into two stages: object detection and pose esti
...
+
+## Results and Models
+
+### COCO Dataset
+
+Results on COCO val2017 with detector having human AP of 56.4 on COCO val2017 dataset
+
+|      Model      | Input Size |  AP   |   AR    |                Details and Download                 |
+| :-------------: | :--------: | :---: | :-----: | :-------------------------------------------------: |
+|     AlexNet     |  256x192   | 0.448 | 0.0.521 |        [cpm_coco.md](./coco/alexnet_coco.md)        |
```
Comment (Ben-Louis) [inline configs/body_2d_keypoint/topdown_heatmap/README.md:21]:
```suggestion
|     AlexNet     |  256x192   | 0.448 | 0.0.521 |        [cpm_coco.md](./coco/alexnet_coco.md)        |
```

File: configs/body_2d_keypoint/topdown_heatmap/README.md [hunk 2]
```diff
@@ -5,3 +5,73 @@ Top-down methods divide the task into two stages: object detection and pose esti
...
+|  SEResNet-101   |  256x192   | 0.734 |  0.790  |     [seresnet_coco.md](./coco/seresnet_coco.md)     |
+|  ShuffleNet-v1  |  256x192   | 0.586 |  0.651  | [shufflenetv1_coco.md](./coco/shufflenetv1_coco.md) |
+|  ShuffleNet-v2  |  256x192   | 0.598 |  0.664  | [shufflenetv2_coco.md](./coco/shufflenetv2_coco.md) |
+|     Swin-T      |  256x192   | 0.724 |  0.782  |         [swin_coco.md](./coco/swin_coco.md)         |
+|     Swin-B      |  256x192   | 0.737 |  0.794  |         [swin_coco.md](./coco/swin_coco.md)         |
+|     Swin-L      |  256x192   | 0.743 |  0.798  |         [swin_coco.md](./coco/swin_coco.md)         |
+
+### MPII Dataset
+
+|     Model      | Input Size | PCKh@0.5 | PCKH@0.1 |                Details and Download                 |
```
Comment (liqikai9) [inline configs/body_2d_keypoint/topdown_heatmap/README.md:59]: `PCKH@0.1` ->  `PCKh@0.1`

File: configs/body_2d_keypoint/topdown_heatmap/README.md [hunk 3]
```diff
@@ -5,3 +5,73 @@ Top-down methods divide the task into two stages: object detection and pose esti
...
+|    ResNet-50    |  256x192   | 0.718 | 0.773 |       [resnet_coco.md](./coco/resnet_coco.md)       |
+|   ResNet-101    |  256x192   | 0.726 | 0.781 |       [resnet_coco.md](./coco/resnet_coco.md)       |
+| ResNet-101+Dark |  256x192   | 0.732 | 0.786 |    [resnet_coco.md](./coco/resnet_dark_coco.md)     |
+|   ResNeSt-50    |  256x192   | 0.720 | 0.775 |      [resnest_coco.md](./coco/resnest_coco.md)      |
+|   ResNeSt-101   |  256x192   | 0.725 | 0.781 |      [resnest_coco.md](./coco/resnest_coco.md)      |
+|  ResNetV1d-50   |  256x192   | 0.722 | 0.777 |    [resnetv1d_coco.md](./coco/resnetv1d_coco.md)    |
+|  ResNetV1d-101  |  256x192   | 0.731 | 0.786 |    [resnetv1d_coco.md](./coco/resnetv1d_coco.md)    |
+|   ResNeXt-50    |  256x192   | 0.715 | 0.771 |      [resnext_coco.md](./coco/resnext_coco.md)      |
+|   ResNeXt-101   |  256x192   | 0.726 | 0.781 |      [resnext_coco.md](./coco/resnext_coco.md)      |
+|     RCN-50      |  256x192   | 0.726 | 0.781 |          [rsn_coco.md](./coco/rsn_coco.md)          |
```
Comment (liqikai9) [inline configs/body_2d_keypoint/topdown_heatmap/README.md:45]: `RCN-50` ->  `RSN-50`, `RCN-50-3x` -> `RSN-50-3x`

File: configs/body_2d_keypoint/topdown_heatmap/README.md [hunk 4]
```diff
@@ -5,3 +5,73 @@ Top-down methods divide the task into two stages: object detection and pose esti
...
+</div>
+
+## Results and Models
+
+### COCO Dataset
+
+Results on COCO val2017 with detector having human AP of 56.4 on COCO val2017 dataset
+
+|      Model      | Input Size |  AP   |  AR   |                Details and Download                 |
+| :-------------: | :--------: | :---: | :---: | :-------------------------------------------------: |
```
Comment (liqikai9) [inline configs/body_2d_keypoint/topdown_heatmap/README.md:20]: `cpm_coco.md` -> `alexnet_coco.md`

File: configs/body_2d_keypoint/topdown_heatmap/README.md [hunk 5]
```diff
@@ -5,3 +5,73 @@ Top-down methods divide the task into two stages: object detection and pose esti
...
+|     AlexNet     |  256x192   | 0.448 | 0.521 |        [cpm_coco.md](./coco/alexnet_coco.md)        |
```
Comment (Ben-Louis) [inline configs/body_2d_keypoint/topdown_heatmap/README.md:21]:
```suggestion
|     AlexNet     |  256x192   | 0.448 | 0.521 |        [alexnet_coco.md](./coco/alexnet_coco.md)        |
```

File: configs/body_2d_keypoint/topdown_heatmap/README.md [hunk 6]
```diff
@@ -5,3 +5,73 @@ Top-down methods divide the task into two stages: object detection and pose esti
...
+|  HRNet-w48+UDP  |  256x192   | 0.773 | 0.820 |    [hrnet_udp_coco.md](./coco/hrnet_udp_coco.md)    |
+|  LiteHRNet-18   |  256x192   | 0.642 | 0.705 |    [litehrnet_coco.md](./coco/litehrnet_coco.md)    |
+|  LiteHRNet-30   |  256x192   | 0.676 | 0.736 |    [litehrnet_coco.md](./coco/litehrnet_coco.md)    |
+|  MobileNet-v2   |  256x192   | 0.647 | 0.708 |  [mobilenetv2_coco.md](./coco/mobilenetv2_coco.md)  |
+|   MSPN 1-stg    |  256x192   | 0.723 | 0.788 |         [mspn_coco.md](./coco/mspn_coco.md)         |
+|   MSPN 4-stg    |  256x192   | 0.765 | 0.826 |         [mspn_coco.md](./coco/mspn_coco.md)         |
+|      PVT-S      |  256x192   | 0.714 | 0.773 |          [pvt_coco.md](./coco/pvt_coco.md)          |
+|    ResNet-50    |  256x192   | 0.718 | 0.773 |       [resnet_coco.md](./coco/resnet_coco.md)       |
+|   ResNet-101    |  256x192   | 0.726 | 0.781 |       [resnet_coco.md](./coco/resnet_coco.md)       |
+| ResNet-101+Dark |  256x192   | 0.732 | 0.786 |    [resnet_coco.md](./coco/resnet_dark_coco.md)     |
```
Comment (Ben-Louis) [inline configs/body_2d_keypoint/topdown_heatmap/README.md:38]:
```suggestion
| ResNet-101+Dark |  256x192   | 0.732 | 0.786 |    [resnet_dark_coco.md](./coco/resnet_dark_coco.md)     |
```

**Comment by Tau-J** [comment]
Do you plan to use this page as an alternative of current algorithm docs? I suggest add it into the docs instead of placing under the config folder. Besides, maybe a accuracy-sorted version is also necessary.

**Comment by ly015** [comment]
> Do you plan to use this page as an alternative of current algorithm docs? I suggest adding it into the docs instead of placing under the config folder. Besides, maybe an accuracy-sorted version is also necessary.

This page is expected to serve as an index for users navigating the configs folder on GitHub. We can modify the doc compiling script to automatically involves these pages in the model zoo document page, or even generate a figure with the table content. That can be a future plan.

As for the sorting, one concern is that the performance comparison may not be completely fair (network scale, input size, ...) so the sorted results could be misleading in some cases. Maybe sort them by the publishing year?

**Comment by Tau-J** [comment]
> > Do you plan to use this page as an alternative of current algorithm docs? I suggest adding it into the docs instead of placing under the config folder. Besides, maybe an accuracy-sorted version is also necessary.
> 
> This page is expected to serve as an index for users navigating the configs folder on GitHub. We can modify the doc compiling script to automatically involves these pages in the model zoo document page, or even generate a figure with the table content. That can be a future plan.
> 
> As for the sorting, one concern is that the performance comparison may not be completely fair (network scale, input size, ...) so the sorted results could be misleading in some cases. Maybe sort them by the publishing year?

I understand your concern. But popular repos like timm and paperwithcodes, whose list is also not completely fair. 
image

image

So this kind of ranking is just for convenience, and you know, users are always wondering "which one is the best model in your repo"

File: configs/body_2d_keypoint/topdown_heatmap/README.md [hunk 7]
```diff
@@ -5,3 +5,107 @@ Top-down methods divide the task into two stages: object detection and pose esti
...
+
+|   Model   | Input Size | PCK(norm. by person size) | PCK (norm. by torso size) |            Details and Download            |
+| :-------: | :--------: | :-----------------------: | :-----------------------: | :----------------------------------------: |
+| ResNet-50 |  256x256   |           96.0            |           80.1            | [resnet_jhmdb.md](./jhmdb/resnet_jhmdb.md) |
+|    CPM    |  368x368   |           89.8            |           65.7            |    [cpm_jhmdb.md](./jhmdb/cpm_jhmdb.md)    |
+
+### PoseTrack2018 Dataset
+
+Results on PoseTrack2018 val with ground-truth bounding boxes.
+| Model | Input Size | AP | Details and Download |
```
Comment (liqikai9) [inline configs/body_2d_keypoint/topdown_heatmap/README.md:108]:
```suggestion
| Model | Input Size | AP | Details and Download |
| :--------: | :--------: | :---: | :----------------------------------------------------: |
```

**Pull request merged**